islandora_rdm_disk_usage/src/Controller/DiskUsageController.php

```diff
@@ -11,226 +11,242 @@
-
-        $report['chart']['content']['#rows'] = $rows;
-        $report['chart']['button'] = array(
-        '#type' => 'button',
-        '#value' => t('switch chart'),
-        '#weight' => 19,
-        '#attributes' => ['id' => 'chart-toggle-button'],
-        '#attached' => array(
-            'library' => array('islandora_rdm_disk_usage/switch-button',),
-        ),
-        //$report['#attached']['library'][] = 'islandora_rdm_disk_usage/switch-button';
-        );
-        // \Kint::$maxLevels = 0;
-        $report['table'] = $this->createTable();
-
-        // Attach js file
-
-
-        return $report;
+      }
+      else {
+        continue;
+      }
     }
-
-    /**
-     * Create a table with the entities. Using Render arrays.
-     */
-    private function createTable() {
-        $entities = $this->getMyList();
-
-        $report['pageTitle'] = [
-            '#type' => 'markup',
-            '#markup' => '<h3>Media Items List</h3>',
-        ];
-
-        $report['table'] = [
-            '#type' => 'table',
-            '#header' => [
-                t('Title'),
-                t('Owner'),
-                t('Size'),
-                t('Type'),
-            ],
-        ];
-
-        // Using array_keys(),array_values()function
-        $key = array_keys($entities);
-        $entity = array_values($entities);
-
-        // Get current page.
-        $page = \Drupal::request()->query->get('page', '');
-        // Starts from 0.
-        $current_page = $page;
-        $num_per_page = 5;
-
-        $begin = $current_page * $num_per_page;
-        $end = $current_page * $num_per_page + $num_per_page;
-
-        if ($end >= count($entities)) {
-            $end = count($entities);
+    $report['table']['#rows'] = $rows;
+
+    // Now that we have the total number of results
+    // Create and initialize the pager.
+    pager_default_initialize(count($entities), $num_per_page);
+
+    // Finally, add the pager to the render array, and return.
+    // Pager_default_initialize sets some global variables so that
+    // The pager theme code 'pager' knows how many page links to create
+    // The pager theme code also create the html code.
+    $report['pager'] = ['#type' => 'pager'];
+
+    return $report;
+  }
+
+  // Create a chart with the entities. Using ChartJS API module.
+
+  /**
+   * Var: $chartType: a string; $groupBy:"owner"/"type".
+   */
+  private function createChart($chartType, $groupBy) {
+    $entities = $this->getMyList();
+
+    // An array to count the size of each media type.
+    $count = [];
+    // Add sizes.
+    foreach ($entities as $key => $entity) {
+
+      if (($entity->hasField('field_file_size'))
+                &&($entity->hasField('revision_user'))
+                &&($entity->hasField('field_mime_type'))) {
+        $size = $entity->get('field_file_size')->value;
+        $owner = $entity->get('revision_user')->getString();
+        $typeFull = $entity->get('field_mime_type')->getString();
```

Comment:
In Drupal we typically only use camelCase for instance variables and methods, this should be in [snake_case](https://en.wikipedia.org/wiki/Snake_case)